aiohttp/connector.py

```diff
@@ -243,6 +256,14 @@ def __init__(
         self._cleanup_closed_transports: List[Optional[asyncio.Transport]] = []
         self._cleanup_closed()
 
+    @cached_property
+    def allowed_protocol_schema_set(self) -> frozenset[str]:
+        """Return allowed protocol schema set.
+
+        By default we allow all protocols for backwards compatibility.
+        """
+        return ALLOWED_PROTOCOL_SCHEMA_SET
```

Comment:
I was thinking about this last night, and maybe the BaseConnector should just have something like:
```
class AllProtocols:
    def __contains__(self, item: object) -> Literal[True]:
        return True
```

We also suggested making this abstract in future, but I think it'd be fine to leave this as allowing anything. Custom connectors can then choose to restrict the allowed set if they choose too.

Reply:
Maybe this is fine as-is and we don't need to worry about removing it long term and can close https://github.com/aio-libs/aiohttp/issues/8633?

Reply:
As you mentioned, this could still be a breaking change if someone has a custom connector that handles other protocols. But, maybe it's fine to just default to http/ws...

Reply:
But, defaulting to something that allows unix/npipe doesn't make sense to me, as there is unlikely to be any connector that supports all of these..

Reply:
So, I'd vote for either:

1. Allow anything and then tighten in subclasses.
2. Allow http/ws and then expand in subclasses.

Reply:
I adjusted it to `BASE_PROTOCOL_SCHEMA_SET`

I think we should merge as-is for backport an than do another PR with a breaking change note to drop 
```
    | TCP_PROTOCOL_SCHEMA_SET
    | UNIX_PROTOCOL_SCHEMA_SET
    | NAMED_PIPE_PROTOCOL_SCHEMA_SET
```

from `BASE_PROTOCOL_SCHEMA_SET` that would only target 4.0

Reply:
OK, I don't see what difference it makes personally. If a custom connector supports another scheme, then this is still a breaking change. If a custom connector supports one of these schemes, then they probably inherited from one of the subclasses, not BaseConnector, and are unaffected by this change regardless.

Reply:
Let's drop it now than